src/Restorer.ts

```diff
@@ -20,15 +20,28 @@ const EVENT_NAME = "restorer:restorefocus";
 const HISOTRY_DEPTH = 10;
 
 class Restorer extends TabsterPart<RestorerProps> implements RestorerInterface {
+    private _observer: MutationObserver | undefined;
+
     constructor(
         tabster: TabsterCore,
         element: HTMLElement,
         props: RestorerProps
     ) {
         super(tabster, element, props);
+
         if (this._props.type === RestorerTypes.Source) {
             const element = this._element?.get();
             element?.addEventListener("focusout", this._onFocusOut);
+
+            // Need to also listen to node removals, which do not trigger focus events in non-Chromium browsers
+            const parent = element?.parentElement;
+            if (parent) {
+                this._observer = new MutationObserver(this._onNodeRemoved);
+                this._observer.observe(parent, {
+                    childList: true,
+                    subtree: false,
```

Comment:
why is subtree false? a focusable node removal could happen anywhere in the tree

Reply:
In this case, there is no need to have a mutation observer per Restorer instance. Tabster already has a mutation observer that will call `dispose` elements with tabster on them. You just need to update the `dispose` function like so 

```ts
dispose(): void {
        if (this._props.type === RestorerTypes.Source) {
            const doc = this._tabster.getWindow().document;
            const element = this._element?.get();
            element?.removeEventListener("focusout", this._onFocusOut);

            if (element && !doc.body.contains(element)) {
                doc.dispatchEvent(
                    new Event(EVENT_NAME, {
                        bubbles: true,
                    })
                );
            }
        }
    }
```

This should fix the issue for all browsers when the element with Restorer it is removed from DOM.

For elements inside the Restorer - the RestorerAPI could have a mutation observer all the Restorer instances (this should probably be opt-in for Restorer through configuration).